server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -124,6 +124,33 @@ public Entry(Snapshot snapshot, boolean includeGlobalState, boolean partial, Sta
             this.version = version;
         }
 
+        private Entry(StreamInput in) throws IOException {
+            snapshot = new Snapshot(in);
+            includeGlobalState = in.readBoolean();
+            partial = in.readBoolean();
+            state = State.fromValue(in.readByte());
+            indices = in.readList(IndexId::new);
+            if (in.getVersion().after(Version.V_4_5_1)) {
+                templates = List.of(in.readStringArray());
+            } else {
+                templates = List.of();
+            }
```

Comment:
This is not a part of commit being backported but it's our own change (https://github.com/crate/crate/pull/11375)